vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,599 @@
+
+The length "vals" in the time_attribute list needs to be the same values as the
+number of columns in the array.
+
+A common unit for netCDF is days since 1900. A full list of accepted time units can
+be found [here](https://ncics.org/portfolio/other-resources/udunits2/).
+
+```{r, TimeAtt, results = 'hide'}
+time_attributes <- list(
+  name = 'time',
+  units = 'days since 1900-01-01',
+  calendar = 'standard',
+  unlim = 'TRUE',
+  vals = time_midpoints # mid point of year
+  )
+```
+
+### Vertical (depth / Z) Attributes
+
+A named list describing the vertical (depth / Z) dimension. The *name*, *units*,
+and *vals* fields are required.
+
+The length of the "vals" in the vertical_attribute list need to match the information
+in the array passed (either the number of cols if there is no time dimension, or the length
+of the third dimension if there is a time dimension).
+
+```{r, VertAtt, results = 'hide'}
+vertical_attributes <- list(
+  name = 'depth',
+  units = 'cm',
+  positive = 'down', # depth is down, height is up
+  vals = depths_max
+)
+```
+
+### CRS Attributes
+
+The CRS attributes describe the coordinate reference system (CRS).
+
+The only required field in the CRS attribute list is the 'proj' field. The proj
+field is the the CRS is proj4string from the PROJ.4 library.
+
+Additionally, we strongly recommend defining the CRS in the crs_wkt format, which
+is preferred by [netCDF convention] (http://cfconventions.org/cf-conventions/cf-conventions.html#use-of-the-crs-well-known-text-format).
+
+The *wkt* function in the raster package can be used to generate wkts.
+
+```{r, CRSAtt, results = 'hide'}
+
+crs_attributes <- list(
+ proj = '+init=epsg:4326',
+  grid_mapping_name = "latitude_longitude",
+  longitude_of_prime_meridian = 0.0,
+  semi_major_axis = 6378137.0,
+  inverse_flattening = 298.257223563
+)
+
+locationsSP <- sp::SpatialPoints(locations)
+sp::proj4string(locationsSP) <- sp::CRS(crs_attributes$proj)
+WKTex <- raster::wkt(locationsSP)
+
+crs_attributes <- c(crs_attributes, crs_wkt = WKTex)
+
+```
+
+### Global Attributes
+
+A named list describing the global attributes of the netCDF. This is where you describe
+things like how the data was made, where the data was made, who the data was made by,
+what climate models were used to create the data, etc.
+
+The *title*, *institution*, *description*, *source_id*, *realm*, *product*, and
+*contact* fields are required.
+
+```{r, GlobalAtt, results = 'hide'}
+global_attributes <- list(
+    title = "Project Name Annual Wet Degree Days",
+    institution = 'Southwest Biological Science Center, U.S. Geological Survey',
+    description = 'methods on how this data was made',
+    source = paste(
+      "SOILWAT2 (v4.2.0);",
+      "rSOILWAT2 (v2.3.2);",
+      "rSFSW2 (v3.1.2)"
+    ),
+    source_id = "SOILWAT2",
+    realm = "land",
+    parent_mip_era = "CMIP5",
+    parent_experiment_id = "RCP45",
+    parent_source = "CanESM2",
+    parent_variant_label = "r1i1p1f1",
```

Comment:
as comment on for the example code in the function documentation:
it is impossible to have a `ripf`-variant for the `CMIP5`-era; variants in the `CMIP5`-era are coded in the format `rip`.